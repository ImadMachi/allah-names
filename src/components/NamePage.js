import React from "react";
import PropTypes from "prop-types";
import { useSelector } from "react-redux";
import { FindNameSelector } from "../features/NamesSlice";
import { Link } from "react-router-dom";

const NamePage = (props) => {
  const id = props.match.params.id;
  const name = useSelector((state) => FindNameSelector(state, id));
  return (
    <div className="name-page">
      <div className="name-page__name">{name.name}</div>
      <p className="name-page__description">{name.description}</p>
      <Link className="name-page__home" to="/">
        العودة إلى القائمة
      </Link>
    </div>
  );
};

NamePage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default NamePage;
