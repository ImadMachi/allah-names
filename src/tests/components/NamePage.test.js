import React from "react";
import NamePage from "./../../components/NamePage";
import { shallow } from "enzyme";

jest.mock("react-redux", () => ({
  useSelector: jest.fn().mockReturnValue({
    name: "الله",
    description: "الاسم الذي يدلّ على ذات الله -تعالى- الجامعة لصفات ألوهيّته.",
    id: "N1",
  }),
}));

test("should render without throwing an error", () => {
  const match = {
    path: "/:id",
    url: "/N1",
    isExact: true,
    params: {
      id: "N1",
    },
  };
  const wrapper = shallow(<NamePage match={match} />);
  expect(wrapper).toMatchSnapshot();
});
