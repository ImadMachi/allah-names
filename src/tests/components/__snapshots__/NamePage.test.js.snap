// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<div
  className="name-page"
>
  <div
    className="name-page__name"
  >
    الله
  </div>
  <p
    className="name-page__description"
  >
    الاسم الذي يدلّ على ذات الله -تعالى- الجامعة لصفات ألوهيّته.
  </p>
  <Link
    className="name-page__home"
    to="/"
  >
    العودة إلى القائمة
  </Link>
</div>
`;
